pkg/api/types.go

```diff
@@ -138,6 +138,21 @@ type ObjectMeta struct {
 	// objects.  Annotation keys have the same formatting restrictions as Label keys. See the
 	// comments on Labels for details.
 	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// List of objects depended by this object. If ALL objects in the list have
+	// been deleted, this object will be garbage collected.
+	OwnerReferences []ObjectReference `json:"ownerReferences,omitempty"`
+
+	// Must be empty before the object is deleted from the registry. The map key
+	// is an ad hoc, explanatory name (e.g., "orphaning finalizer"), the value
+	// is an identifier for the responsible component that will remove the entry
+	// from the map. If the ObjectMeta.DeletionTimestamp is non-nil, entry in
+	// this map can only be removed.
+	Finailzers map[string]string `json:"finalizers,omitempty"`
+
+	// Should the dependents be orphaned when this object is deleted. Default to
```

Comment:
If we are still going to keep this field, please change comments to start like

"OrphanDependents teels if dependents should..."